pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1595,241 +1621,120 @@ func parseConfigItems(config *zconfig.EdgeDevConfig, ctx *getconfigContext) {
 	// Start with the defaults so that we revert to default when no data
 	newGlobalConfig := types.GlobalConfigDefaults
 
+	// Reset ConfigItem Errors
+	globalStatusPtr := &ctx.zedagentCtx.globalStatus
```

Comment:
The go routine which is sending device info messages to zedcloud might run concurrently with this parse code.
Hence instead of clearing globalStatus up front it makes sense to have a newGlobalStatus, and then at the end of the function update zedagentCtx.globalStatus. That avoids zedcloud seeing transient state due to the parser running concurrently.

Reply:
Makes sense.. Will do that.
Do we need locking here to be consistent? 
  Acquire lock before Updating the Ptr?
  Acquire lock when sending deviceInfo messages?

Reply:
Atomic assignment (or near atomic if we update the struct) should be sufficient. Sprinkling locks leads to a mess based on my experience.